evalml/tests/preprocessing_tests/test_split_data.py

```diff
@@ -59,3 +59,97 @@ def test_split_data(
             y = pd.Series(y)
         pd.testing.assert_frame_equal(X_test, X[int(train_size) :], check_dtype=False)
         pd.testing.assert_series_equal(y_test, y[int(train_size) :], check_dtype=False)
+
+
+@pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
+@pytest.mark.parametrize("data_type", ["np", "pd", "ww"])
+def test_split_data_defaults(
+    problem_type,
+    data_type,
+    X_y_binary,
+    X_y_multi,
+    X_y_regression,
+    make_data_type,
+):
+    if is_binary(problem_type):
+        X, y = X_y_binary
+    if is_multiclass(problem_type):
+        X, y = X_y_multi
+    if is_regression(problem_type):
+        X, y = X_y_regression
+    problem_configuration = None
+    if is_time_series(problem_type):
+        problem_configuration = {"gap": 1, "max_delay": 7, "time_index": "ts_data"}
+
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
```

Comment:
I suspect this is boiler plate not just for these two tests but also across many of the tests.  Did you want to refactor it into something that returns `problem_configuration`, `X`, and `y` given `data_type`, `problem_type`, etc?  I am pretty sure I've seen this a few other places too.  I think it's also fine to file an issue to call out the refactor with the modules we should refactor this code into.

Reply:
I like this. A slight refactor to `get_test_data_from_configuration` works wonderfully to factor this out! We can go through and replace other instances in a future story.